tests/test_metadata_handling.py

```diff
@@ -0,0 +1,67 @@
+import unittest
+import yaml
+from shaker import helpers
+from nose.tools import raises
+
+class TestMetadataHandling(unittest.TestCase):
+
+    # Sample metadata with duplicates
+    _sample_metadata_duplicates = {
+        "dependencies": [
+            "git@github.com:test_organisation/test1-formula.git==v1.0.1",
+            "git@github.com:test_organisation/test1-formula.git==v1.0.1",
```

Comment:
It should have had a different version, the effect is the same though, the first one is dropped (theres no higher version resolutions or anything fancy) The versions are just there to illustrate which is kept. We could add something more 'intelligent' later but this is just to avoid a mess of duplicates overwriting each other